samples-generator/lib/create3dtilesBatchTableExt.js

```diff
@@ -0,0 +1,116 @@
+var cesium3dTilesBatch = 'CESIUM_3dtiles_batch_table';
+
+function sortByByteOffset(a, b) {
+    if (a.byteOffset < b.byteOffset ){
+        return -1;
+    }
+
+    if (a.byteOffset < b.byteOffset) {
+        return 1;
+    }
+
+    return 0;
+}
+
+/**
+ * @param {Object} gltf The gltf object to edit
+ * @param {Object} humanAttributes  Key value object, where each key corresponds to the name of the
+ *                                  attribute, and each value corresponds to a non-binary array
+ *                                  of values corresponding to this key. e.g {"Names": ["A", "B"]}
+ *                                  These will be stored directly in the `properties`
+ *                                  section of the modified GLTF.
+ * @param {Object} binaryAttributes Key value object, where each key corresponds to the name of the
+ *                                  binary attribute, and the value in object with the following keys:
+ *                                  'byteLength', 'byteOffset', 'componentType', 'type'
+ *
+ * @param {Buffer} sharedBinaryBuffer Currently the samples generator only supports creating
+ *                                    ONE additional Buffer with all the binary attributes inside of it.
+ *                                    Each value in `binaryAttributes` must correspond to this buffer. It
+ *                                    is an error to call this function using binaryAttributes without also
+ *                                    providing this value.
+ */
+
+function create3dtilesBatchTableExt(gltf, humanAttributes, binaryAttributes, sharedBinaryBuffer) {
+    gltf['extensionsUsed'] = [cesium3dTilesBatch];
+
+    if (!defined(humanAttributes) && !defined(binaryAttributes)) {
+        throw new Error('humanAttributes or binaryAttributes must be defined');
+    }
+
+    if (defined(binaryAttributes) && !defined(sharedBinaryBuffer)) {
+        throw new Error('sharedBinaryBuffer must be defined if binaryAttributes is defined');
+    }
+
+    if (!defined(gltf['extensionsUsed'])) {
+        gltf['extensionsUsed'] = [];
+    }
+
+    if (!defined(gltf['extensions'])) {
+        gltf['extensions'] = {};
+    }
+
+    gltf['extensions'][cesium3dTilesBatch] = { batchTables: [] };
+
+    var humanAttributeNames = Object.keys(humanAttributes);
+    var i=0;
+
+    var newBufferIndex = gltf.buffers.length;
+
+    if (defined(humanAttributes)) {
+        var batchLength = humanAttributeNames.length > 0 ? humanAttributes[humanAttributeNames[0]].length : 0;
+        var newBatchTable = {batchLength: batchLength, properties: {}};
+
+        for (i = 0; i < humanAttributeNames.length; ++i) {
+            var values = humanAttributes[humanAttributeNames[i]];
+            // TODO: Should we hardcode a if batchAttributes[i] === 'id' here?
+            //       It could fail if we have an batch attribute named id that doesn't use
+            //       an implicit indexing structure (i.e 2, 4, 8, 1, 9, 30...);
```

Comment:
I think it's working fine, the id property in the glTF output looked correct to me.